src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -51,7 +52,16 @@ public async Task<IActionResult> Assign([FromBody] EncryptedHostAssignmentContex
                 return StatusCode(StatusCodes.Status500InternalServerError, error);
             }
 
-            var result = _instanceManager.StartAssignment(assignmentContext);
+            var result = false;
+            if (string.Equals(assignmentContext.ZipUrlEnvVar, EnvironmentSettingNames.ScmRunFromPackage,
+                StringComparison.OrdinalIgnoreCase))
+            {
+                result = _instanceManager.SpecializeForScmBuilds(assignmentContext);
```

Comment:
See if you can handle all this in StartAssignment(). Looks like we can get rid of all changes in this file